### PR TITLE
Fix Docs for Modules and Refactor `@publicModule` tag to `@exportedAs`

### LINKED ISSUE #1258: dgeni output: need a way to provide a description at the module level
Currently, all the text at the top-level for a module is autogenerated. 
screen shot 2015-03-30 at 9 58 55 pm

We want a way to specify a comment block to use for the text. For example, angular2/annotations.

### PR DESCRIPTION
**Pull request by petebacondarwin**
@naomiblack and @mhevery
- this PR fixes the module documentation issue. Now you can provide a jsdoc style comment block at the top of the module definition file that must include the `@module` tag to indicate that it documents the module rather than code components exported from the module.  Also you now specify that a module is or public consumption by applying the `@public` tag on the module itself. For example:

``` js
/**
 * @module
 * @public
 * @description
 * This block contains markdown that describe this module.
 */
```

The other major change in this PR is that the `@publicModule` tag has been replaced with the `@exportedAs` tag, which can be applied multiple times to a single code component. What this implies is that the associated code component has also been exported on other modules.

Closes #1258

### CODE REVIEW

File: modules/angular2/change_detection.js [hunk 1]
```diff
@@ -45,7 +52,7 @@ export var defaultPipes = {
 
 
 /**
- * @publicModule angular2/change_detection
+ * @exportedFrom angular2/change_detection
```
Comment (mhevery) [inline modules/angular2/change_detection.js]: Why the rename? I think `publicModule` is a better name. Can you share your thoughts?

Comment (mhevery) [inline modules/angular2/change_detection.js]: I think it should be `exportedAs`

**Comment by mhevery** [comment]
Rename `exportedFrom` to `exportedAs`

**Comment by petebacondarwin** [comment]
The reason for the rename is that it no longer puts the onus on the exported component to tell that the module is public. This is also considerably more flexible as it allows modules that only contain direct exports and not reexports to be public. Also it allows items to be decl

**Comment by petebacondarwin** [comment]
Phone finger slippage...

Also it allows components to be reexported internally too for the private docs

**Comment by petebacondarwin** [comment]
Regarding the naming...

The tag is attached to a component that is an export on a private module. The tag declares that this component is also exported **from** another module. Exported **as** implies that a module is being aliased, which is not accurate.

But if you are sure you want it that way then I can change it.

**Comment by mhevery** [comment]
I think `exportedAs` is still right. As it is being aliased.

Misko Hevery [10:03 AM]
it will be available from two places, it is just that we will not publish the original place.

**Comment by petebacondarwin** [comment]
@mhevery - updated with `@exportedAs`. Feel free to get someone to merge once the rename PR has gone in.

**Pull request closed without merging**